src/main/java/org/elasticsearch/index/query/CommonTermsQueryBuilder.java

```diff
@@ -217,4 +292,136 @@ public void doXContent(XContentBuilder builder, Params params) throws IOExceptio
     public String queryId() {
         return NAME;
     }
+
+    @Override
+    public Query toQuery(QueryParseContext parseContext) throws QueryParsingException, IOException {
+        String field;
+        FieldMapper mapper = parseContext.fieldMapper(fieldName);
+        if (mapper != null) {
+            field = mapper.names().indexName();
+        } else {
+            field = fieldName;
+        }
+
+        Analyzer analyzerObj;
+        if (analyzer == null) {
+            if (mapper != null) {
+                analyzerObj = parseContext.getSearchAnalyzer(mapper);
+            } else {
+                analyzerObj = parseContext.mapperService().searchAnalyzer();
+            }
```

Comment:
I am not clear on the changes made here, the previous code block was:
```
            if (mapper != null) {
                analyzer = mapper.fieldType().searchAnalyzer();
            }
            if (analyzer == null && mapper != null) {
                analyzer = parseContext.getSearchAnalyzer(mapper);
            }
            if (analyzer == null) {
                analyzer = parseContext.mapperService().searchAnalyzer();
            }
```

I am not sure after the refactoring the result will be the same, take into account that the method calls can return null, it is not the same as an else block, if that makes sense.